metadata/src/audio/item.rs

```diff
@@ -16,98 +17,268 @@ use librespot_core::{
 
 pub type AudioItemResult = Result<AudioItem, Error>;
 
-// A wrapper with fields the player needs
 #[derive(Debug, Clone)]
 pub struct AudioItem {
-    pub id: SpotifyId,
-    pub spotify_uri: String,
+    pub track_id: SpotifyId,
+    pub uri: String,
     pub files: AudioFiles,
     pub name: String,
-    pub duration: i32,
+    pub covers: Vec<String>,
```

Comment:
This is the only minor concern I have with the `AudioItem` definition:

While it's definitely sensible to have the IDs serialized into URLs (for which you need a reference to the session, which isn't always available), we do lose some IMO valuable information: the sizes of the images. While sorting the images by size does help a bit in this direction, maybe we can somehow keep the original sizes in the list?

One example solution:

```rust
struct ImageUrl {
    url: String,
    size: ImageSize,
    width: i32,
    height: i32,
}
```

Reply:
There is already an Image struct that is pretty close to identical that maybe we could modify so we don't have almost duplicate structs?

Reply:
I assume you're referring to [`metadata::image::Image`](https://github.com/librespot-org/librespot/blob/16dbade51665667ecf768f21099afc648e0b0176/metadata/src/image.rs#L17-L23)? Indeed, that is the one that I took for reference. The problem here is that you'll need a reference to the `Session` to retrieve the `image-url` and you can't do that inside a `From<&ImageMessage>`.

Reply:
Yep. I know we need session. It was just a thought. I'll see what I can come up with.

Reply:
Done. I called it CoverImage but the fields are the same.